### PR TITLE
Refresh AI landing and add new form

### PR DESCRIPTION
**Pull request by mtruj013**
## Done

- Bumped Vanilla version to 4.1 
- Applied content and design changes to landing page as per [doc](https://docs.google.com/document/d/1HP9YdHF11yVQ9SLl0VXhLezDZREsr1GHIKNi9wT1Gv0/edit) and [mock-up](https://www.figma.com/file/sGeCjFvA1kV9Kmvg2S2VBn/23.10-U.com---AI-bubble?node-id=248%3A1446&mode=dev)
- Built new form in the new style as per [copy doc](https://docs.google.com/document/d/1maC3tCmTAgY1G_0nd0aAzw5ZMNeZqbkyawCRMG5Rgxk/edit) and [mock-up](https://www.figma.com/file/sGeCjFvA1kV9Kmvg2S2VBn/23.10-U.com---AI-bubble?node-id=764%3A9597&mode=dev)

## QA

- View the site locally in your web browser at: https://ubuntu-com-13138.demos.haus/ai
- Check against design and doc
- Make sure all CTAs work as expected
- Fill out the form and make sure it submits to marketo

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-5577, https://warthogs.atlassian.net/browse/WD-5547

### CODE REVIEW

**Comment by mtruj013** [comment]
@lyubomir-popov continuing from the discussion [here](https://warthogs.atlassian.net/browse/WD-5873?focusedCommentId=308884), I marked this page as ready for design review within the last 20 minutes, but I see your comments are from about an hour ago. If that's correct then you reviewed a WIP version of this page, I think the bulk of your comments have been addressed if you could take a look and confirm?

**Comment by juanruitina** [comment]
- [ ] Logos should fit in a row, but I guess this has to do with the whole logo cloud kerfuffle.
- [ ] Fix the capitalisation here: "Read more about Ubuntu workstations ›"
- [ ] The spacing between the HR and the link at the bottom of the grid of Charmed apps (e.g. What is Kubeflow ›" is too tight. I think just wrapping the link on a `p` element would do?
- [ ] Also here, can we please increase the space between for each Charmed app on mobile, or remove the spacing between paragraph and HR+link? It gets confusing there, where one ends and the other starts. Maybe @lyubomir-popov can assist with this.

Screenshot 2023-09-07 at 13 22 46

- [ ] The image contain doesn't align with the grid, it's wider. Also, the gray background goes beyond the illustration, at least at the bottom.
- [ ] Please remove chevron after "Case study: Real-time space tracking with AI supercomputing"

**Comment by lyubomir-popov** [comment]
One more, sorry! The links here need to be in paragraphs - so theyu get the padding-top/margin-bottom 
image

**Comment by lyubomir-popov** [comment]
@mtruj013 re-reviewing, can you please make the logos 128px (8rem) tall:
image

this should be top-aligned:
image

ca nwe please avoid this gap:
image

The spacing around this (top/bottom) is a bit much, ca nwe please have u-no-margin--bottom on the buttons, u-no-padding--bottom on all strips/sections.

**Comment by mtruj013** [comment]
@juanruitina @lyubomir-popov thanks both! I've addressed your comments if you could take another look?

Also about the Canonical + Nvidia logo being top aligned, the extra spacing on the top comes from the asset itself, is another version that can be used? 
image

**Comment by juanruitina** [comment]
Great job, LGTM!

**Comment by lyubomir-popov** [comment]
@mtruj013 some tiny spacing things still appear in the pr:

- buttons can't be in p's - it interferes with the spacing:

image

- this p-section class is not needed - we need to get rid of this padding:
- 
image

- p-section still has u-no-margin, instead of u-no-padding--bottom, so the padding is still there:
(together, these 3 things add about 100px of extra white space and push content unneceserily) 
image

- small overlap here on medium screens:
image

- the orange tag still not attached to the hr above it:
image

The rest looks really nice, thank you for the hard work on this.

**Comment by mtruj013** [comment]
Thanks @lyubomir-popov, I'll add the fixes, but for the Canonical + Nivida logo please see my [comment above](https://github.com/canonical/ubuntu.com/pull/13138#issuecomment-1711704505)

**Comment by mtruj013** [comment]
@lyubomir-popov that spacing distorts the image a bit
image
so I adjusted slightly 
image

anyway, ready for another look!

**Comment by lyubomir-popov** [comment]
@mtruj013 you're right, 41 is thecorrect size!

**Comment by juanruitina** [comment]
Forgot to QA the form, my apologies. Please make sure to add the semi-transparent overlay under it. I also think some more padding is needed around it, particularly the top one.

File: templates/shared/forms/interactive/ai.html [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                <label class="p-checkbox js-checkbox">
+                  <input type="checkbox" aria-labelledby="notSure" class="p-checkbox__input">
+                  <span class="p-checkbox__label" id="notSure">Not sure</span>
+                </label>
+              </div>
+              <div class="p-section--shallow">
+                <hr class="p-rule">
+                <label for="mlopsRequirements">Add details about your AI or MLOps requirements</label>
+                <textarea id="mlopsRequirements"></textarea>
+
```
Comment (carkod) [inline templates/shared/forms/interactive/ai.html:48]: There is one line added by Sophie missing that says "Add your information"?

File: package.json [hunk 2]
```diff
@@ -115,7 +115,7 @@
     "typescript": "4.5.4",
     "url-polyfill": "1.1.12",
     "url-search-params-polyfill": "8.1.1",
-    "vanilla-framework": "^3.14.0",
+    "vanilla-framework": "4.1.0",
```
Comment (carkod) [inline package.json:118]: wow aren't there any breaking changes that could be happening with this?

**Review by carkod — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (mtruj013) [reply package.json:118]: We can't really rebrand bubbles while not being on V4. (I mean we can, but then we'd kind of have to do the work twice cause at some point all the custom styles I'd have to write to make it _look_ like it's on V4 would need to removed). But yes, regardless, this needs to be thoroughly QA'd. Relevant discussion [here](https://chat.canonical.com/canonical/pl/wi8kg469sjrp5r1utprbf4b9fo)

File: templates/shared/forms/interactive/ai.html (hunk 1 above)
Comment (mtruj013) [reply templates/shared/forms/interactive/ai.html]: @Sophie-32 @juanruitina @lyubomir-popov in my opinion it looks better without that line. It looks a bit out of place. See:
Screenshot from 2023-09-11 09-48-46
 
vs

image


If we have to keep it, what's the ask from a design perspective (it wasn't included in the mock-up) to make it look a bit better? An hr on top? Lmk what you all think is best and I'll implement ASAP!

Comment (Sophie-32) [reply templates/shared/forms/interactive/ai.html]: I'd prefer to keep it, but will keep this in mind for future forms. I defer to @lyubomir-popov on adding the HR, but that seems like the best option from my perspective.

**Comment by lyubomir-popov** [comment]
@mtruj013 the lines are there to indicate there are 3 subsecitons in the form, I'd rather keep them.

One thing - this px hardcoding doesn't work on larger screens - as the value of 1rem changes to 1.125 rem. It should use the $grid-max-width variable (or a rem based hardcoded equivalent)./ Otherwise, it becomes narrower than the grid on bigger screens (>1680):

image

**Comment by mtruj013** [comment]
>  the lines are there to indicate there are 3 subsecitons in the form, I'd rather keep them.

@lyubomir-popov I'm not asking about the lines, I'm asking about the extra line of copy "Add your information" that wasn't included in the design. From the mock-up: 
image

from the doc (the pink bit is what's missing): 
image


So my question was how to style this since it wasn't included in the mock-up. Should I just add an hr on top? It currently looks like this:
image

**Comment by mtruj013** [comment]
@lyubomir-popov I've updated the max-width variable, please let me know if this was what you meant

**Comment by lyubomir-popov** [comment]
@mtruj013 I meant set the max-width (not min-width, this might break small screens) to this vanilla variable:
https://github.com/canonical/vanilla-framework/blob/eefce1c418b22c117ce8aeaf802dc58dceebf1eb/scss/_settings_grid.scss#L5

currently, it is even narower (875px):
image


you'll have to factor in the dialog padding and also poosition the close button absolutely - otherwise the grid in the dialog is still misaligned with the grid on the regular page:
image

**Comment by mtruj013** [comment]
Update: We've had a call and decided to merge as is, we will tackle any additional changes to the modal in a separate PR

**Pull request merged**